iOS Example Frame/iOS Example Frame/ViewController/HomeViewController.swift

```diff
@@ -170,12 +165,50 @@ class HomeViewController: UIViewController {
 
 extension HomeViewController: ThreedsWebViewControllerDelegate {
   func threeDSWebViewControllerAuthenticationDidSucceed(_ threeDSWebViewController: ThreedsWebViewController, token: String?) {
-      threeDSWebViewController.dismiss(animated: true, completion: nil)
-      showAlert(with: "3DS success, token: \(token ?? "nil")")
+    threeDSWebViewController.dismiss(animated: true, completion: nil)
+    showAlert(with: "\(token ?? "nil")", title: "3DS success")
   }
 
   func threeDSWebViewControllerAuthenticationDidFail(_ threeDSWebViewController: ThreedsWebViewController) {
-      threeDSWebViewController.dismiss(animated: true, completion: nil)
-      showAlert(with: "3DS Fail")
+    threeDSWebViewController.dismiss(animated: true, completion: nil)
+    showAlert(with: "3DS Fail")
+  }
+}
+
+extension HomeViewController: PKPaymentAuthorizationViewControllerDelegate {
+
+  func paymentAuthorizationViewControllerDidFinish(_ controller: PKPaymentAuthorizationViewController) {
+    dismiss(animated: true, completion: nil)
+  }
+
+  func paymentAuthorizationViewController(_ controller: PKPaymentAuthorizationViewController,
+                                          didAuthorizePayment payment: PKPayment,
+                                          handler completion: @escaping (PKPaymentAuthorizationResult) -> Void) {
+    dismiss(animated: true, completion: nil)
+
+    // pull test payload from static file
+    guard let paymentDataURL = Bundle.main.url( forResource: "example_apple_pay_payment_data", withExtension: "json") else {
+      print("Unable to get URL of Apple Pay payment data.")
+      return
+    }
+    let paymentData: Data
+
+    do {
+      paymentData = try Data(contentsOf: paymentDataURL)
+    } catch {
+      print(error.localizedDescription)
+      return
+    }
+    let applePay = ApplePay(tokenData: paymentData)
+    checkoutAPIService.createToken(.applePay(applePay)) { result in
+      DispatchQueue.main.async { [weak self] in
+        switch result {
+          case .success(let tokenDetails):
+            self?.showAlert(with: tokenDetails.token, title: "Success")
+          case .failure(let error):
+            self?.showAlert(with: error.localizedDescription, title: "Error")
+        }
+      }
+    }
```

Comment:
This method should do the minimum (dismiss the Payment screen - surprised its not already; pass payment data)

```swift
dismiss(animated: true)
processToken(payment.token.paymentData)
```